src/math.F90

```diff
@@ -659,4 +675,90 @@ function watt_spectrum(a, b) result(E_out)
 
   end function watt_spectrum
 
+!===============================================================================
+! W acts as a front end to the MIT Faddeeva function, Faddeeva_w.
+!===============================================================================
+
+  subroutine w(Z,wv)
+    use ISO_C_BINDING
+    complex(C_DOUBLE_COMPLEX), intent(inout) :: Z ! The point to evaluate Z at
+    complex(8), intent(out) :: wv                 ! The resulting w(Z) value
+    real(C_DOUBLE) :: relerr ! Target relative error in inner loop of MIT Faddeeva
+    logical        :: mangle ! Do we need to perform the special mangling step?
+
+    ! There is some special mangling the true W function in WHOPPER does.  I
+    ! replicate it here.
```

Comment:
This actually brought up an interesting problem.  I am actually evaluating F(z), the Doppler-broadened line shape functions.  The transformation in W_WHOPPER between F(z) and W(z) is given as {F(z)  = W(z), im(z) > 0 || F(z) = -W*(z*), im(z) < 0}.  I have replicated this in the code.

The issue is, the integral F(z) is actually equal to {F(z)  = W(z), im(z) > 0 || F(z) = W*(z*), im(z) < 0} (note the missing negative in the second part).  I have no idea where this sign change came from, but although Hwang's papers aren't exactly carefully written, WHOPPER itself was.  I will have to track down where this issue came from.

On the bright side, this sign flip barely changes the answer.